n-acd: fix constants from the RFC

bengal opened the pull request:
Fix the following constants from RFC 5227:

 - PROBE_MAX is 2 seconds instead of 3
 - ANNOUNCE_NUM is 2 instead of 3

Therefore, the maximum timeout for the probe is 7 seconds instead of 9. Also update the related constants.

bengal committed:
n-acd: fix constants from the RFC

Fix the following constants from RFC 5227:

 - PROBE_MAX is 2 seconds instead of 3
 - ANNOUNCE_NUM is 2 instead of 3

Therefore, the maximum timeout for the probe is 7 seconds instead of
9. Also update the related constants.

dvdhrm reviewed (approved):
How did we get this wrong? Ugh! This is obviously correct, and it also matches what ipv4acd.c does in libsystemd-network (which kinda inspired this code).

bengal committed:
n-acd: avoid truncation of the timeout jitter

When the timeout multiplier is big enough (for example, the value
corresponding to a total timeout of 20 seconds), the jitter value can
become larger than ~4.3 seconds and its value in nanoseconds doesn't
fit into the unsigned int. This leads to a truncation which reduces
the actual computed jitter.

Use a 64-bit unsigned int to avoid this problem.

Signed-off-by: Beniamino Galvani <bgalvani@redhat.com>

dvdhrm reviewed (approved):
Looks good, thanks!
